pycqed/measurement/waveform_control/pulse_library.py

```diff
@@ -63,6 +63,27 @@ def __init__(self, name, element_name, I_channel, Q_channel,
         self.alpha = alpha
         self.phi_skew = phi_skew
 
+    @classmethod
```

Comment:
I don't like very much the necessity to add all these paraneters and defautls here, as it duplicates the information in the init function.

You can retrieve the arguments and their default values from the definition of the init function via:
```
import inspect
inspect.signature(cls.__init__)
```

This would allow defining this function only once in the base class

Reply:
Example:

```
class Foo:
    def __init__(self, a, *args, b=1, **kw):
        pass
    
import inspect
sig = inspect.signature(Foo.__init__)

print(sig.parameters.keys())

par = sig.parameters['a']
print(par.kind, par.default)

par = sig.parameters['b']
print(par.kind, par.default)

par = sig.parameters['args']
print(par.kind, par.default)

par = sig.parameters['kw']
print(par.kind, par.default)
```
Output:
> odict_keys(['self', 'a', 'args', 'b', 'kw'])
> POSITIONAL_OR_KEYWORD <class 'inspect._empty'>
> KEYWORD_ONLY 1
> VAR_POSITIONAL <class 'inspect._empty'>
> VAR_KEYWORD <class 'inspect._empty'>

Reply:
Addressed by https://github.com/QudevETH/PycQED_py3/pull/62/commits/5094f90da89c37a43e150e21f4bff14c02e32497